### ISSUE TITLE
Can't build Angular 5-beta+2 docs under linux

### ISSUE THREAD

**Issue by chalin** [dart-lang/dartdoc#1728]
**Environment:**

- Dart VM version: 2.0.0-dev.69.0 (Tue Jul 17 14:57:16 2018 +0200) on "linux_x64"
- dartdoc version: 0.20.1

**Log:**

[This Travis log](https://travis-ci.org/dart-lang/site-webdev/jobs/406496410) contains the failure:

```console
[19:57:45] Dartdoc for packages: acx,forms,ng,router,test.
[19:57:45] > pub global run dartdoc --version
[19:57:45] dartdoc version: 0.20.1
...
[20:02:41] EXEC start: ...
  > pub global run dartdoc --include=angular,angular.security {"cwd":"tmp/pub-packages/angular-5.0.0-beta+2"}
[20:02:42] Generating documentation for 'angular' into /home/travis/build/dart-lang/site-webdev/tmp/pub-packages/angular-5.0.0-beta+2/doc/api/
...
[20:03:58] Generating docs for library angular from package:angular/angular.dart...

[20:03:59] 
Generation failed: Unsupported operation: 
package:dartdoc/src/model.dart 1261:40                                                           Dynamic.enclosingElement
package:dartdoc/src/model.dart 2822:9                                                            ModelElement._buildCanonicalModelElement
package:dartdoc/src/model.dart 2833:34                                                           ModelElement.canonicalModelElement
package:dartdoc/src/markdown_processor.dart 275:27                                               _getMatchingLinkElement
package:dartdoc/src/markdown_processor.dart 726:12                                               _linkDocReference
package:dartdoc/src/markdown_processor.dart 978:26                                               Documentation._renderMarkdownToHtml._linkResolver
package:markdown/src/inline_parser.dart 699:14                                                   LinkSyntax._resolveReferenceLink
package:markdown/src/inline_parser.dart 721:9                                                    LinkSyntax._tryAddReferenceLink
package:markdown/src/inline_parser.dart 671:12                                                   LinkSyntax.onMatchEnd
package:markdown/src/inline_parser.dart 1191:16                                                  TagState.close
package:markdown/src/inline_parser.dart 1130:7                                                   TagState.tryMatch
package:markdown/src/inline_parser.dart 104:57                                                   InlineParser.parse.<fn>
dart:_internal                                                                                   ListIterable.any
package:markdown/src/inline_parser.dart 104:12                                                   InlineParser.parse
package:dartdoc/src/markdown_processor.dart 872:57                                               MarkdownDocument._parseInlineContent
package:dartdoc/src/markdown_processor.dart 877:9                                                MarkdownDocument._parseInlineContent
package:dartdoc/src/markdown_processor.dart 897:5                                                MarkdownDocument.renderLinesToHtml
package:dartdoc/src/markdown_processor.dart 988:21                                               Documentation._renderMarkdownToHtml
package:dartdoc/src/markdown_processor.dart 958:9                                                Documentation._renderHtmlForDartdoc
package:dartdoc/src/markdown_processor.dart 943:7                                                Documentation.asOneLiner
package:dartdoc/src/model.dart 3209:29                                                           ModelElement.oneLineDoc
dart:mirrors                                                                                     _LocalInstanceMirror.getField
package:mustache4dart/src/mirrors.dart 111:35                                                    _MethodMirrorField.val
package:mustache4dart/mustache_context.dart 122:45                                               _MustacheContext._getMustacheContext
package:mustache4dart/mustache_context.dart 118:12                                               _MustacheContext._getContextForKey
package:mustache4dart/mustache_context.dart 82:18                                                _MustacheContext._getInThisOrParent
package:mustache4dart/mustache_context.dart 78:12                                                _MustacheContext.field
/home/travis/.pub-cache/hosted/pub.dartlang.org/mustache4dart-2.1.2/lib/src/tokens.dart 169:21   _ExpressionToken.apply
/home/travis/.pub-cache/hosted/pub.dartlang.org/mustache4dart-2.1.2/lib/src/tokens.dart 40:18    Token.call
/home/travis/.pub-cache/hosted/pub.dartlang.org/mustache4dart-2.1.2/lib/src/tmpl.dart 101:20     _Template._write
/home/travis/.pub-cache/hosted/pub.dartlang.org/mustache4dart-2.1.2/lib/src/tmpl.dart 76:5       _Template.call
/home/travis/.pub-cache/hosted/pub.dartlang.org/mustache4dart-2.1.2/lib/src/mustache.dart 11:64  render
/home/travis/.pub-cache/hosted/pub.dartlang.org/mustache4dart-2.1.2/lib/src/tokens.dart 211:16   _PartialToken.apply
/home/travis/.pub-cache/hosted/pub.dartlang.org/mustache4dart-2.1.2/lib/src/tokens.dart 286:57   _StartSectionToken.apply.<fn>.<fn>
/home/travis/.pub-cache/hosted/pub.dartlang.org/mustache4dart-2.1.2/lib/src/tokens.dart 309:8    _StartSectionToken.forEachUntilEndSection
/home/travis/.pub-cache/hosted/pub.dartlang.org/mustache4dart-2.1.2/lib/src/tokens.dart 286:9    _StartSectionToken.apply.<fn>
dart:core                                                                                        Iterable.forEach
/home/travis/.pub-cache/hosted/pub.dartlang.org/mustache4dart-2.1.2/lib/src/tokens.dart 285:27   _StartSectionToken.apply
/home/travis/.pub-cache/hosted/pub.dartlang.org/mustache4dart-2.1.2/lib/src/tokens.dart 299:53   _StartSectionToken.apply.<fn>
/home/travis/.pub-cache/hosted/pub.dartlang.org/mustache4dart-2.1.2/lib/src/tokens.dart 309:8    _StartSectionToken.forEachUntilEndSection
/home/travis/.pub-cache/hosted/pub.dartlang.org/mustache4dart-2.1.2/lib/src/tokens.dart 299:5    _StartSectionToken.apply
/home/travis/.pub-cache/hosted/pub.dartlang.org/mustache4dart-2.1.2/lib/src/tokens.dart 40:18    Token.call
/home/travis/.pub-cache/hosted/pub.dartlang.org/mustache4dart-2.1.2/lib/src/tmpl.dart 101:20     _Template._write
/home/travis/.pub-cache/hosted/pub.dartlang.org/mustache4dart-2.1.2/lib/src/tmpl.dart 76:5       _Template.call
package:dartdoc/src/html/html_generator_instance.dart 288:30                                     HtmlGeneratorInstance._build
package:dartdoc/src/html/html_generator_instance.dart 183:5                                      HtmlGeneratorInstance.generateLibrary
package:dartdoc/src/html/html_generator_instance.dart 94:9                                       HtmlGeneratorInstance._generateDocs
package:dartdoc/src/html/html_generator_instance.dart 36:7                                       HtmlGeneratorInstance.generate
===== asynchronous gap ===========================
dart:async                                                                                       _AsyncAwaitCompleter.completeError
package:dartdoc/src/html/html_generator_instance.dart                                            HtmlGeneratorInstance.generate
===== asynchronous gap ===========================
dart:async                                                                                       _asyncThenWrapperHelper
package:dartdoc/dartdoc.dart                                                                     Dartdoc.generateDocs
/home/travis/.pub-cache/hosted/pub.dartlang.org/dartdoc-0.20.1/bin/dartdoc.dart 76:46            main.<fn>.<fn>
```

I can't reproduce it under macOS.

cc @kwalrath

**Comment by jcollins-g** [comment]
I can reproduce this.  Looking into it now.

**Comment by jcollins-g** [comment]
This is a problem with inconsistent special object handling (at least).  #1554

**Comment by chalin** [comment]
@jcollins-g - after downgrading dartdoc to version 0.20.0 I was able to generate the docs. I assume that we can use 0.20.0 for the next little while (in case you have more pressing issues to address)?

**Comment by jcollins-g** [comment]
0.20.0 is shortly going to stop working >dev-69.0.  So this is pretty high on my list to get fixed.

**Comment by jcollins-g** [comment]
And strangely, despite fixing the problem I saw and testing to make sure that this exception didn't happen, after merging I see it again.

**Comment by jcollins-g** [comment]
Might be related to the SDK version -- I'm using 69.2 post merge and 69.0 is what I tested the fix against.

**Comment by jcollins-g** [comment]
Angular was broken with two problems -- one that I fixed in #1729, and this one.  It looks like this issue only occurs when `[dynamic]` is part of a comment reference, and only with an updated version of the analyzer/SDK.

**Comment by jcollins-g** [comment]
@chalin Give recently published version 0.20.2 a spin, it should fix your problem.

**Comment by chalin** [comment]
@jcollins-g: I confirm that it is [working now](https://travis-ci.org/dart-lang/site-webdev/jobs/408147734), thanks!
